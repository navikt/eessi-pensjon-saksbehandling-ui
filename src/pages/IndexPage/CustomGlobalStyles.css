#root {
  --ac-button-secondary-bg: white;
}
