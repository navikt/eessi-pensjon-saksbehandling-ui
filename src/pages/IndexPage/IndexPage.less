.frontPage {
    .frontPageLink {
      transform: translateX(-20px);
      opacity: 0;
      animation: frontPage-slideInFromLeft 0.2s forwards;
    }
}
@keyframes frontPage-slideInFromLeft {
  0% {
    opacity: 0;
    transform: translateX(-20px);
  }
  100% {
    opacity: 1;
    transform: translateX(0);
  }
}
