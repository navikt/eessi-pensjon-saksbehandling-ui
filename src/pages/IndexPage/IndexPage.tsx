import { readNotification } from 'actions/pagenotification'
import ContextBanner from 'components/ContextBanner/ContextBanner'
import Modal from 'components/Modal/Modal'
import TopContainer from 'components/TopContainer/TopContainer'
import { BUCMode, FeatureToggles } from 'declarations/app'
import { State } from 'declarations/reducers'
import { timeLogger } from 'metrics/loggers'
import Dashboard, { LayoutTabs, Widgets } from '@navikt/dashboard'
import { BodyLong } from '@navikt/ds-react'
import { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useDispatch, useSelector } from 'react-redux'
import styled from 'styled-components/macro'
import * as extraWidgets from 'widgets'
import { VerticalSeparatorDiv } from '@navikt/hoykontrast'

const CustomDashboard = styled(Dashboard)`
   flex: 1 1 auto;
   position: relative;
   margin: 0 auto;
   width: 100%;
   padding: 0rem 1rem;
`

export interface IndexPageProps {
  username?: string
}

const defaultLayouts: LayoutTabs = [{
  label: 'default',
  body: {
    lg: [
      { i: 'w-1-overview', x: 0, y: 0, w: 12, h: 2, minW: 6, maxW: 12, minH: 2, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 1, w: 12, h: 6, minW: 6, maxW: 12, minH: 2, maxH: 999 }
    ],
    md: [
      { i: 'w-1-overview', x: 0, y: 0, w: 3, h: 2, minW: 2, maxW: 3, minH: 2, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 1, w: 3, h: 6, minW: 2, maxW: 3, minH: 2, maxH: 999 }
    ],
    sm: [
      { i: 'w-1-overview', x: 0, y: 0, w: 1, h: 2, minW: 1, maxW: 1, minH: 2, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 1, w: 1, h: 6, minW: 1, maxW: 1, minH: 2, maxH: 999 }
    ]
  }
}]

const defaultLayoutsWithOthers: LayoutTabs = [{
  label: 'default',
  body: {
    lg: [
      { i: 'w-1-overview', x: 0, y: 0, w: 12, h: 2, minW: 6, maxW: 12, minH: 2, maxH: 999 },
      { i: 'w-3-journalføring', x: 0, y: 1, w: 4, h: 1, minW: 2, maxW: 12, minH: 1, maxH: 999 },
      { i: 'w-4-pagenotification', x: 4, y: 1, w: 4, h: 1, minW: 2, maxW: 12, minH: 1, maxH: 999 },
      { i: 'w-5-s3inventory', x: 8, y: 1, w: 4, h: 1, minW: 2, maxW: 12, minH: 1, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 2, w: 12, h: 6, minW: 6, maxW: 12, minH: 1, maxH: 999 }
    ],
    md: [
      { i: 'w-1-overview', x: 0, y: 0, w: 3, h: 2, minW: 2, maxW: 3, minH: 2, maxH: 999 },
      { i: 'w-3-journalføring', x: 0, y: 1, w: 1, h: 1, minW: 1, maxW: 3, minH: 1, maxH: 999 },
      { i: 'w-4-pagenotification', x: 1, y: 1, w: 1, h: 1, minW: 1, maxW: 3, minH: 1, maxH: 999 },
      { i: 'w-5-s3inventory', x: 2, y: 1, w: 1, h: 1, minW: 1, maxW: 3, minH: 1, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 2, w: 3, h: 6, minW: 2, maxW: 3, minH: 2, maxH: 999 }
    ],
    sm: [
      { i: 'w-1-overview', x: 0, y: 0, w: 1, h: 2, minW: 1, maxW: 1, minH: 2, maxH: 999 },
      { i: 'w-3-journalføring', x: 0, y: 1, w: 1, h: 1, minW: 1, maxW: 1, minH: 1, maxH: 999 },
      { i: 'w-4-pagenotification', x: 0, y: 2, w: 1, h: 1, minW: 1, maxW: 1, minH: 1, maxH: 999 },
      { i: 'w-5-s3inventory', x: 0, y: 3, w: 1, h: 1, minW: 1, maxW: 1, minH: 1, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 4, w: 1, h: 6, minW: 1, maxW: 1, minH: 2, maxH: 999 }
    ]
  }
}]

const defaultWidgets: Widgets = [{
  i: 'w-1-overview',
  type: 'overview',
  title: 'Overview widget',
  visible: true,
  options: {
    collapsed: true
  }
}, {
  i: 'w-2-buc',
  type: 'buc',
  title: 'BUC widget',
  visible: true,
  options: {
    allowFullScreen: true
  }
}, {
  i: 'w-3-journalføring',
  type: 'journalføring',
  title: 'Journalføring widget',
  visible: true,
  options: {
    collapsed: true
  }
}, {
  i: 'w-4-pagenotification',
  type: 'pagenotification',
  title: 'Page notification widget',
  visible: true,
  options: {
    collapsed: true
  }
}, {
  i: 'w-5-s3inventory',
  type: 's3inventory',
  title: 'S3 inventory widget',
  visible: true,
  options: {
    collapsed: true
  }
}]

const defaultConfig = {
  cols: { lg: 12, md: 3, sm: 1 },
  breakpoints: { lg: 992, md: 768, sm: 0 },
  margin: [10, 10],
  containerPadding: [10, 10],
  rowHeight: 30,
  defaultTabIndex: 0,
  version: 1
}

export interface IndexPageSelector {
  featureToggles: FeatureToggles
  mode: BUCMode
  username: string | undefined
  message: string | null | undefined
  show: boolean | undefined
  byline: string | null | undefined
}

const mapState = (state: State): IndexPageSelector => ({
  featureToggles: state.app.featureToggles,
  mode: state.buc.mode,
  username: state.app.username,
  message: state.pagenotification.message,
  show: state.pagenotification.show,
  byline: state.pagenotification.byline

})

export const IndexPage: React.FC<IndexPageProps> = (): JSX.Element => {
  const { featureToggles, mode, message, show, byline }: IndexPageSelector = useSelector<State, IndexPageSelector>(mapState)
  const dispatch = useDispatch()
  const { t } = useTranslation()
  const [displayedMessage, setDisplayedMessage] = useState<boolean>(false)
  const [showModal, setShowModal] = useState<boolean>(false)
  const [loggedTime] = useState<Date>(new Date())

  useEffect(() => {
    dispatch(readNotification())
  }, [])

  useEffect(() => {
    if (!displayedMessage && message) {
      setShowModal(true)
      setDisplayedMessage(true)
    }
  }, [message, displayedMessage])

  useEffect(() => {
    return () => {
      timeLogger('view', loggedTime)
    }
  }, [loggedTime])

  const allowedWidgets = featureToggles.ADMIN_NOTIFICATION_MESSAGE
    ? ['buc', 'overview', 'journalføring', 'pagenotification', 's3inventory']
    : ['buc', 'overview']

  return (
    <TopContainer>
      <Modal
        open={showModal && !!show}
        modal={{
          modalTitle: t('ui:notification'),
          modalContent: (
            <div style={{ padding: '2rem', minWidth: '400px', textAlign: 'center' }}>
              <BodyLong>
                {message}
              </BodyLong>
              <VerticalSeparatorDiv />
              <BodyLong>
                {byline}
              </BodyLong>
            </div>
          ),
          modalButtons: [{
            main: true,
            text: 'OK',
            onClick: () => setShowModal(false)
          }]
        }}
        onModalClose={() => setShowModal(false)}
      />
      <ContextBanner mode={mode} />
      <CustomDashboard
        id='eessi-pensjon-ui-fss'
        configurable
        extraWidgets={extraWidgets}
        defaultWidgets={defaultWidgets}
        defaultLayouts={featureToggles.ADMIN_NOTIFICATION_MESSAGE ? defaultLayoutsWithOthers : defaultLayouts}
        defaultConfig={defaultConfig}
        allowedWidgets={allowedWidgets}
      />
    </TopContainer>
  )
}

export default IndexPage
