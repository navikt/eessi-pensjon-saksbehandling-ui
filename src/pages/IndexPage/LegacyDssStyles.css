#root,
.ReactModalPortal {
  /* Grå ramme rundt flagg */
  --navds-semantic-color-border: var(--navds-global-color-gray-600);
  --navds-global-color-gray-600: rgba(112, 112, 112, 1);

  /* Grå bakgrunnsfarge i annenhver tabellrad */
  --navds-semantic-color-canvas-background: var(--navds-global-color-gray-100);
  --navds-semantic-color-component-background-light: var(--navds-global-color-white);
  --navds-global-color-gray-100: rgba(241, 241, 241, 1);
  --navds-global-color-white: rgba(255, 255, 255, 1);
}
