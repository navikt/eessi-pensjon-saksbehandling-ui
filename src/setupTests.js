import Enzyme, { shallow, render, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
// import { expect } from 'chai'
import cheerio from 'cheerio'

Enzyme.configure({ adapter: new Adapter() })

global.shallow = shallow
global.render = render
global.mount = mount

global.cheerio = cheerio
