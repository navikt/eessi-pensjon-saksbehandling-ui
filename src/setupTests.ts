import Enzyme, { shallow, render, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { act } from 'react-dom/test-utils'
Enzyme.configure({ adapter: new Adapter() });

(global as any).shallow = shallow;
(global as any).render = render;
(global as any).mount = mount;
(global as any).act = act

// eslint-disable-next-line no-undef
HTMLCanvasElement.prototype.getContext = jest.fn()
window.scrollTo = jest.fn()
window.location.reload = jest.fn()
window.fetch = require('jest-fetch-mock')
Object.defineProperty(window, 'getComputedStyle', {
  value: () => ({
    getPropertyValue: () => ('')
  })
})
