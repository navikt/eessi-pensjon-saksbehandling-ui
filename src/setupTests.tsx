
import { render, fireEvent, screen, getNodeText, cleanup } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { useDispatch, useSelector } from 'react-redux'
import 'jest-styled-components'
import '@testing-library/jest-dom'
import { TextEncoder, TextDecoder } from 'util';
(global as any).TextEncoder = TextEncoder;
(global as any).TextDecoder = TextDecoder;

jest.mock('react-redux');

(global as any).screen = screen;
(global as any).render = render;
(global as any).getNodeText = getNodeText;
(global as any).fireEvent = fireEvent;
(global as any).cleanup = cleanup;
(global as any).act = act

HTMLCanvasElement.prototype.getContext = jest.fn()
window.scrollTo = jest.fn()

class ResizeObserver {
  observe () {}
  unobserve () {}
  disconnect () {}
}
window.ResizeObserver = ResizeObserver

const crypto = require('crypto')

Object.defineProperty(global.self, 'crypto', {
  value: {
    getRandomValues: (arr: any) => crypto.randomBytes(arr.length)
  }
})

Object.defineProperty(window, 'getComputedStyle', {
  value: () => ({
    getPropertyValue: () => ('')
  })
})

jest.mock('uuid', () => ({
  v4: () => 'mock-uuid'
}))

jest.mock('i18next', () => {
  const use = jest.fn()
  const init = jest.fn()
  const loadLanguages = jest.fn()
  const t = jest.fn()
  const result = {
    use,
    init,
    loadLanguages,
    t
  }
  use.mockImplementation(() => result)
  return result
})

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

jest.mock('react-pdf', () => {
  return {
    pdfjs: { GlobalWorkerOptions: { workerSrc: '' } },
    Document: (props: any) => {
      setTimeout(() => {
        props.onLoadSuccess({ numPages: 5 })
      }, 500)
      return (
        <div className='mock-pdfdocument'>
          {props.children}
          </div>
      )
    },
    Page: (props: any) => {
      return (
        <div className='mock-pdfpage'>
          {'Page: '}{props.pageNumber}
      </div>
    )
    }
  }
})

export const stageSelector = (defaultSelector: any, params: any) => {
  (useDispatch as jest.Mock).mockImplementation(() => jest.fn());
  (useSelector as jest.Mock).mockImplementation(() => ({
    ...defaultSelector,
    ...params
  }))
}
