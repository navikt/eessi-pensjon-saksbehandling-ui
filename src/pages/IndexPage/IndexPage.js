import React from 'react'
import PT from 'prop-types'
import ReactTooltip from 'react-tooltip'
import { connect } from 'store'
import { withTranslation } from 'react-i18next'
import TopContainer from 'components/TopContainer/TopContainer'
import { Dashboard } from 'eessi-pensjon-ui'
import * as extraWidgets from 'widgets'
import './IndexPage.css'

const defaultLayout = [{
  label: 'default',
  body: {
    lg: [
      { i: 'w-1-overview', x: 0, y: 0, w: 12, h: 1, minW: 6, maxW: 12, minH: 1, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 2, w: 12, h: 6, minW: 6, maxW: 12, minH: 2, maxH: 999 }
    ],
    md: [
      { i: 'w-1-overview', x: 0, y: 0, w: 3, h: 1, minW: 2, maxW: 3, minH: 1, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 2, w: 3, h: 6, minW: 2, maxW: 3, minH: 2, maxH: 999 }
    ],
    sm: [
      { i: 'w-1-overview', x: 0, y: 0, w: 1, h: 1, minW: 1, maxW: 1, minH: 1, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 2, w: 1, h: 6, minW: 1, maxW: 1, minH: 2, maxH: 999 }
    ]
  }
}]

const defaultLayoutWithVarsel = [{
  label: 'default',
  body: {
    lg: [
      { i: 'w-1-overview', x: 0, y: 0, w: 12, h: 1, minW: 6, maxW: 12, minH: 1, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 2, w: 12, h: 6, minW: 6, maxW: 12, minH: 2, maxH: 999 },
      { i: 'w-3-varsler', x: 0, y: 8, w: 12, h: 2, minW: 6, maxW: 12, minH: 2, maxH: 999 }
    ],
    md: [
      { i: 'w-1-overview', x: 0, y: 0, w: 3, h: 1, minW: 2, maxW: 3, minH: 1, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 2, w: 3, h: 6, minW: 2, maxW: 3, minH: 2, maxH: 999 },
      { i: 'w-3-varsler', x: 0, y: 8, w: 3, h: 2, minW: 2, maxW: 3, minH: 2, maxH: 999 }

    ],
    sm: [
      { i: 'w-1-overview', x: 0, y: 0, w: 1, h: 1, minW: 1, maxW: 1, minH: 1, maxH: 999 },
      { i: 'w-2-buc', x: 0, y: 2, w: 1, h: 6, minW: 1, maxW: 1, minH: 2, maxH: 999 },
      { i: 'w-3-varsler', x: 0, y: 8, w: 1, h: 2, minW: 1, maxW: 1, minH: 2, maxH: 999 }
    ]
  }
}]

const defaultWidgets = [{
  i: 'w-1-overview',
  type: 'overview',
  title: 'Overview widget',
  visible: true,
  options: {
    collapsed: true
  }
}, {
  i: 'w-2-buc',
  type: 'buc',
  title: 'BUC widget',
  visible: true,
  options: {}
}, {
  i: 'w-3-varsler',
  type: 'varsler',
  title: 'Varsler widget',
  visible: true,
  options: {}
}]

const defaultConfig = {
  cols: { lg: 12, md: 3, sm: 1 },
  breakpoints: { lg: 900, md: 600, sm: 0 },
  margin: [10, 10],
  containerPadding: [10, 10],
  rowHeight: 30,
  defaultTabIndex: 0,
  version: 1
}

const allowedWidgets = ['buc', 'varsler', 'overview']

const mapStateToProps = (state) => ({
  username: state.app.username
})

export const IndexPage = ({ history, t, username }) => {
  const afterLayoutChange = () => {
    ReactTooltip.rebuild()
  }
  return (
    <TopContainer
      className='p-indexPage'
      t={t}
      history={history}
    >
      <ReactTooltip place='top' type='dark' effect='solid' />
      <Dashboard
        id='eessi-pensjon-ui-fss'
        extraWidgets={extraWidgets}
        defaultWidgets={defaultWidgets}
        defaultLayout={username === 'Z990706' ? defaultLayoutWithVarsel : defaultLayout}
        defaultConfig={defaultConfig}
        allowedWidgets={allowedWidgets}
        afterLayoutChange={afterLayoutChange}
      />
    </TopContainer>
  )
}

IndexPage.propTypes = {
  history: PT.object.isRequired,
  t: PT.func.isRequired,
  username: PT.string
}

export default connect(mapStateToProps, () => {})(withTranslation()(IndexPage))
