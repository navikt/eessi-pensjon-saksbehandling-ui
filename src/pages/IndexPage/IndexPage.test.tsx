import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import ReactTooltip from 'react-tooltip'
import { stageSelector } from 'setupTests'
import { IndexPage, IndexPageProps, IndexPageSelector } from './IndexPage'

ReactTooltip.rebuild = jest.fn()

const defaultSelector: IndexPageSelector = {
  username: 'mockUsername'
}

jest.mock('components/TopContainer/TopContainer', () => {
  return ({ children }: {children: JSX.Element}) => (
    <div className='mock-c-topcontainer'>
      {children}
    </div>
  )
})

jest.mock('eessi-pensjon-ui', () => {
  const Ui = jest.requireActual('eessi-pensjon-ui').default
  return {
    ...Ui,
    Dashboard: (props: any) => <div className='mock-c-dashboard' onClick={() => props.afterLayoutChange()} />,
  }
})


describe('pages/IndexPage', () => {
  let wrapper: ReactWrapper
  const initialMockProps: IndexPageProps = {
    history: {}
  }

  beforeAll(() => {
    stageSelector(defaultSelector, {})
  })

  beforeEach(() => {
    wrapper = mount(<IndexPage {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    // expect(wrapper).toMatchSnapshot()  // Until React.Tooltip has uuid
  })

  it('IndexPage has proper HTML structure', () => {
    expect(wrapper.exists('.p-indexPage')).toBeTruthy()
    expect(wrapper.exists('.mock-c-dashboard')).toBeTruthy()
  })

  it('Layout change triggers a tooltip rebuild', () => {
    wrapper.find('.mock-c-dashboard').simulate('click')
    expect(ReactTooltip.rebuild).toHaveBeenCalled()
  })
})
