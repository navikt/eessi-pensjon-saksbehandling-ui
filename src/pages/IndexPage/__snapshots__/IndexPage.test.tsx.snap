// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/IndexPage Render: match snapshot 1`] = `
.c1 {
  max-width: 1440px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
  margin: 0 auto;
  width: 100%;
  padding: 0rem 1rem;
}

ui-fss.g226[id="c0"] {
  content: "c1,";
}

<IndexPage>
  <Component>
    <div
      className="mock-c-topcontainer"
    >
      <IEAlert
        highContrast={false}
      >
        <div />
      </IEAlert>
      <Component
        highContrast={false}
        mode="buclist"
      >
        <div
          className="mock-c-contextbanner"
        />
      </Component>
      <Styled(Component)
        allowedWidgets={
          Array [
            "buc",
            "overview",
          ]
        }
        configurable={true}
        defaultConfig={
          Object {
            "breakpoints": Object {
              "lg": 992,
              "md": 768,
              "sm": 0,
            },
            "cols": Object {
              "lg": 12,
              "md": 3,
              "sm": 1,
            },
            "containerPadding": Array [
              10,
              10,
            ],
            "defaultTabIndex": 0,
            "margin": Array [
              10,
              10,
            ],
            "rowHeight": 30,
            "version": 1,
          }
        }
        defaultLayouts={
          Array [
            Object {
              "body": Object {
                "lg": Array [
                  Object {
                    "h": 1,
                    "i": "w-1-overview",
                    "maxH": 999,
                    "maxW": 12,
                    "minH": 1,
                    "minW": 6,
                    "w": 12,
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "h": 6,
                    "i": "w-2-buc",
                    "maxH": 999,
                    "maxW": 12,
                    "minH": 2,
                    "minW": 6,
                    "w": 12,
                    "x": 0,
                    "y": 2,
                  },
                ],
                "md": Array [
                  Object {
                    "h": 1,
                    "i": "w-1-overview",
                    "maxH": 999,
                    "maxW": 3,
                    "minH": 1,
                    "minW": 2,
                    "w": 3,
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "h": 6,
                    "i": "w-2-buc",
                    "maxH": 999,
                    "maxW": 3,
                    "minH": 2,
                    "minW": 2,
                    "w": 3,
                    "x": 0,
                    "y": 2,
                  },
                ],
                "sm": Array [
                  Object {
                    "h": 1,
                    "i": "w-1-overview",
                    "maxH": 999,
                    "maxW": 1,
                    "minH": 1,
                    "minW": 1,
                    "w": 1,
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "h": 6,
                    "i": "w-2-buc",
                    "maxH": 999,
                    "maxW": 1,
                    "minH": 2,
                    "minW": 1,
                    "w": 1,
                    "x": 0,
                    "y": 2,
                  },
                ],
              },
              "label": "default",
            },
          ]
        }
        defaultWidgets={
          Array [
            Object {
              "i": "w-1-overview",
              "options": Object {
                "collapsed": true,
              },
              "title": "Overview widget",
              "type": "overview",
              "visible": true,
            },
            Object {
              "i": "w-2-buc",
              "options": Object {
                "allowFullScreen": true,
              },
              "title": "BUC widget",
              "type": "buc",
              "visible": true,
            },
          ]
        }
        extraWidgets={
          Object {
            "BUCWidget": [Function],
            "OverviewWidget": [Function],
          }
        }
        highContrast={false}
        id="eessi-pensjon-ui-fss"
      >
        <Component
          allowedWidgets={
            Array [
              "buc",
              "overview",
            ]
          }
          className="c0 c1"
          configurable={true}
          defaultConfig={
            Object {
              "breakpoints": Object {
                "lg": 992,
                "md": 768,
                "sm": 0,
              },
              "cols": Object {
                "lg": 12,
                "md": 3,
                "sm": 1,
              },
              "containerPadding": Array [
                10,
                10,
              ],
              "defaultTabIndex": 0,
              "margin": Array [
                10,
                10,
              ],
              "rowHeight": 30,
              "version": 1,
            }
          }
          defaultLayouts={
            Array [
              Object {
                "body": Object {
                  "lg": Array [
                    Object {
                      "h": 1,
                      "i": "w-1-overview",
                      "maxH": 999,
                      "maxW": 12,
                      "minH": 1,
                      "minW": 6,
                      "w": 12,
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "h": 6,
                      "i": "w-2-buc",
                      "maxH": 999,
                      "maxW": 12,
                      "minH": 2,
                      "minW": 6,
                      "w": 12,
                      "x": 0,
                      "y": 2,
                    },
                  ],
                  "md": Array [
                    Object {
                      "h": 1,
                      "i": "w-1-overview",
                      "maxH": 999,
                      "maxW": 3,
                      "minH": 1,
                      "minW": 2,
                      "w": 3,
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "h": 6,
                      "i": "w-2-buc",
                      "maxH": 999,
                      "maxW": 3,
                      "minH": 2,
                      "minW": 2,
                      "w": 3,
                      "x": 0,
                      "y": 2,
                    },
                  ],
                  "sm": Array [
                    Object {
                      "h": 1,
                      "i": "w-1-overview",
                      "maxH": 999,
                      "maxW": 1,
                      "minH": 1,
                      "minW": 1,
                      "w": 1,
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "h": 6,
                      "i": "w-2-buc",
                      "maxH": 999,
                      "maxW": 1,
                      "minH": 2,
                      "minW": 1,
                      "w": 1,
                      "x": 0,
                      "y": 2,
                    },
                  ],
                },
                "label": "default",
              },
            ]
          }
          defaultWidgets={
            Array [
              Object {
                "i": "w-1-overview",
                "options": Object {
                  "collapsed": true,
                },
                "title": "Overview widget",
                "type": "overview",
                "visible": true,
              },
              Object {
                "i": "w-2-buc",
                "options": Object {
                  "allowFullScreen": true,
                },
                "title": "BUC widget",
                "type": "buc",
                "visible": true,
              },
            ]
          }
          extraWidgets={
            Object {
              "BUCWidget": [Function],
              "OverviewWidget": [Function],
            }
          }
          highContrast={false}
          id="eessi-pensjon-ui-fss"
        >
          <div
            className="mock-c-dashboard"
            onClick={[Function]}
          />
        </Component>
      </Styled(Component)>
    </div>
  </Component>
</IndexPage>
`;
